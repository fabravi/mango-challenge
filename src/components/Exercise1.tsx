"use client";
import Range from "@/ui/Range";

import { useState } from "react";

interface Exercise1Props {
  min: number;
  max: number;
}

export default function Exercise1({ min, max }: Exercise1Props) {
  const [value, setValue] = useState({ min, max });

  return (
    <div>
      <Range min={min} max={max} value={value} onChange={setValue} />
    </div>
  );
}
