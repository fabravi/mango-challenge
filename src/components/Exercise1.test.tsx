import { render } from "@testing-library/react";
import Exercise1 from "./Exercise1";

describe("Exercise1", () => {
  test("should render correctly", () => {
    const { container } = render(<Exercise1 min={0} max={100} />);
    expect(container).toMatchSnapshot();
  });
});
