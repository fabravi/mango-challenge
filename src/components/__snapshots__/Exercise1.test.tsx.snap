// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exercise1 should render correctly 1`] = `
<div>
  <div>
    <div
      class="container"
      role="slider"
    >
      <fieldset
        class="fieldset"
      >
        <label
          class=""
          for="min"
        >
          €
        </label>
        <input
          class="input "
          data-testid="min"
          max="100"
          min="0"
          type="number"
          value="0.00"
        />
      </fieldset>
      <div
        class="range"
      >
        <span
          class="rail"
        />
        <span
          class="track"
          role="slider-track"
          style="left: 0%; width: 100%;"
        />
        <span
          class="thumb "
          role="slider-thumb"
          style="left: 0%;"
          tabindex="0"
        />
        <span
          class="thumb "
          role="slider-thumb"
          style="left: 100%;"
          tabindex="0"
        />
      </div>
      <fieldset
        class="fieldset"
      >
        <label
          class=""
          for="max"
        >
          €
        </label>
        <input
          class="input "
          data-testid="max"
          max="100"
          min="0"
          type="number"
          value="100.00"
        />
      </fieldset>
    </div>
  </div>
</div>
`;
